### PR TITLE
Add QuickCheck

### LINKED ISSUE #2: Update the tests to use QuickCheck

### PR DESCRIPTION
**Pull request by eriknstevenson**
Fixes #2.

Happy Hacktoberfest!

### CODE REVIEW

File: Crypto/Enigma.hs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 module Crypto.Enigma where
 
 import Crypto.Enigma.Helper
+import Test.QuickCheck (Arbitrary(..), elements, shuffle, vectorOf)
```
Comment (kc1212) [inline Crypto/Enigma.hs]: Is it possible to move this to the Test.hs file?

File: Crypto/Enigma.hs [hunk 2]
```diff
@@ -13,6 +14,13 @@ data Conf = Conf
     , getRing :: [Char]}
     deriving (Show)
 
+instance Arbitrary Conf where
+    arbitrary =
+        Conf <$> return plugs
+             <*> elements [refB, refC]
+             <*> (fmap (take 3) $ shuffle [rtypeI, rtypeII, rtypeIII, rtypeIV, rtypeV])
+             <*> vectorOf 3 (elements plugs)
+
```
Comment (kc1212) [inline Crypto/Enigma.hs]: See the comment above

**Review by kc1212 — CHANGES_REQUESTED** [review]

File: enigma-hs.cabal [hunk 3]
```diff
@@ -19,16 +19,17 @@ library
   ghc-options:         -Wall
   exposed-modules:     Crypto.Enigma, Crypto.Enigma.Helper
   exposed:             True
-  -- other-modules:       
-  -- other-extensions:    
-  build-depends:       base >=4.6
+  -- other-modules:
+  -- other-extensions:
+  build-depends:       base >=4.6,
+                       QuickCheck
```
Comment (kc1212) [inline enigma-hs.cabal]: Ideally only QuickCheck should be in the test-suite, but I think the cabal file is a bit screwed up, e.g. the executable is in the tests folder. You're welcome to fix it, but that's not necessary for the issue at hand.

**Comment by eriknstevenson** [comment]
Thanks for taking a look!

I just moved the instance to `Test.hs` and wrapped it up in a newtype to avoid having an orphan instance.

**Review by kc1212 — APPROVED** [review]

**Pull request merged**